bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/connection/KNXConnection.java

```diff
@@ -313,20 +367,26 @@ private static KNXNetworkLink connectBySerial(String serialPort) throws KNXExcep
 
     /*
      * (non-Javadoc)
-     * 
+     *
      * @see org.osgi.service.cm.ManagedService#updated(java.util.Dictionary)
      */
     @Override
     public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+
+        while (!bundleActivated) {
+            try {
+                sLogger.debug("updated, waiting for bundle activation.");
+                Thread.sleep(500);
+            } catch (InterruptedException e1) {
+                // TODO Auto-generated catch block
+                e1.printStackTrace();
+            }
+        }
```

Comment:
we'll have to remove this busy waiting here since the configuration lifecycle of the whole OSGi container get's stuck by this. What exactly is your intention? Please have a look into the new configuration mechanism being shown in the current archetype (or some of the newer bindings). They doesn't use the ManagedService anymore which give more control about the state of the configuration while activating the binding.